### DISCUSSION TITLE
Powering esp8266 from GPS port.

### DISCUSSION THREAD

**Discussion by agaletski** [syssi/esphome-jk-bms#641] [Q&A] [answered]
Is it possible and safe to use available  GND and VCC output of GPS port to power esp8286 module ?
I know that  step-down dc dc is necessary to provide 3.3v. My question is about current limit on this port.
Thanks for any info.

**Comment by agaletski** [comment, marked as answer]
Tested the issue myself and now I can confirm GPS port's Vcc & GND leads  of JK BMS provide enough power to feed ESP8266 module. Which is excellent as we can combine all components in a single box and connect it to BMS by single cable with appropriate terminal. Active ESP8266 module itself consumes 70mA current (3.3v). GPS socket provides Vcc close to battery voltage (26-27v in my case) I used LM2596 module (tuned it to get 3.3v output before connect)

**Reply by syssi** [reply]
Please be careful because of the common ground. It's recommended to power the ESP isolated. You have to make sure there is no high current flow via the GND pin of the GPS port if the MOSFET turns off.

**Reply by agaletski** [reply]
Make ESP module properly isolated assumes using optocouplers for serial connections too. It is good but makes extra complications for schema.
Because of really low current consumption (about 20-30 mA on BMS terminal) Im sure, feeding module from GPS port and direct connection to serial TTL lines is an acceptable solution.

**Reply by syssi** [reply]
You have been warned. There is the risk to kill your setup (BMS or ESP) as soon the Mosfet is turned off.

**Reply by meccip** [reply]
i can confirm that you are playing with fire. i been there and burn the gps output. At start up, buck converter create a spark and could be fatal to gps input. But i was using bms at 50v, lets say that maybe is working at 24v. I try powering from LCD output, where is 12v, but after some time bms is shuting off without any reason and i have to turn it on. In present i use 20v laptop charger connected to battery(50v) on input AC and i have isolated 20v power supply and then step-down to 5v (working with no issues for half year)
